### PR TITLE
Fix build and test suite

### PR DESCRIPTION
**Pull request by remicollet**

### CODE REVIEW

File: config.m4 [hunk 1]
```diff
@@ -15,30 +15,28 @@ if test "$PHP_PHP_TRIE" != "no"; then
...
   fi
 
   if test -z "$HATTRIE_LIB"; then
     AC_MSG_RESULT(hattrie lib is not properly installed. You will not be able to use the HAT trie)
-    dnl AC_MSG_ERROR(Please install the hattrie library)
+    AC_MSG_ERROR(Please install the hattrie library)
   fi
 
   dnl add support for C++
-  CXXFLAGS="-std=c++11"
```
Comment (remicollet) [inline config.m4:34]: breaks system build flags

File: config.m4 [hunk 2]
```diff
@@ -15,30 +15,28 @@ if test "$PHP_PHP_TRIE" != "no"; then
...
   PHP_REQUIRE_CXX()
+  PHP_ADD_INCLUDE($HATTRIE_LIB)
```
Comment (remicollet) [inline config.m4:35]: Instead of harcoded path in code

File: config.m4 [hunk 3]
```diff
@@ -15,30 +15,28 @@ if test "$PHP_PHP_TRIE" != "no"; then
...
 
-  PHP_SUBST(PHP_TRIE_SHARED_LIBADD)
```
Comment (remicollet) [inline config.m4:37]: This is not a shared library (only a header one)

File: config.m4 [hunk 4]
```diff
@@ -15,30 +15,28 @@ if test "$PHP_PHP_TRIE" != "no"; then
...
   
   dnl add C++11 standard library
-  PHP_ADD_LIBRARY(stdc++, 1, PHP_TRIE_SHARED_LIBADD)
```
Comment (remicollet) [inline config.m4:40]: always there for cxx build

File: trie.h [hunk 5]
```diff
@@ -15,7 +15,7 @@
 #include <utility>
 #include <cstddef>
 #include <string>
-#include <hat-trie/include/tsl/htrie_map.h>
+#include <htrie_map.h>
```
Comment (cmb69) [inline trie.h:18]: This needs to be conditionally included, since support for hattrie is optional.

Comment (remicollet) [reply trie.h:18]: I think conditional build without hattrie is not ready ;)

Comment (cmb69) [reply trie.h:18]: Too bad.

**Comment by rvencu** [comment]
just as a comment, I had to manually create the hat-trie folder inside /root/tmp/pear/php_trie in order to get the compilation succeed. There was no option not to use HAT

I am using a cPanel server and did installation from PECL module installers

Before this location I tried all lib locations I could find (/usr/lib /usr/local/lib, lib64 versions, /opt/cpanel/php-73/root/usr/lib and so on and on)

**Pull request merged**